plugin/trino-resource-group-managers/src/main/java/io/trino/plugin/resourcegroups/db/DbResourceGroupsModule.java

```diff
@@ -33,4 +35,11 @@ public void configure(Binder binder)
         binder.bind(ResourceGroupConfigurationManager.class).to(DbResourceGroupConfigurationManager.class).in(Scopes.SINGLETON);
         newExporter(binder).export(DbResourceGroupConfigurationManager.class).withGeneratedName();
     }
+
+    @Singleton
+    @Provides
+    public FlywayMigration flywayMigration(DbResourceGroupConfig config)
```

Comment:
This can use normal injection. Just add `@Inject` on the constructor and bind it as a singleton.

Reply:
What we should add here is something to ensure the database is migrated before it is used. To do this create an empty interface:
```java
@SuppressWarnings("MarkerInterface")
public interface DatabaseReady {}
```
Then make `FlywayMigration` implement it and bind it directly and to the interface:
```java
binder.bind(FlywayMigration.class).asEagerSingleton();
binder.bind(DatabaseReady.class).to(FlywayMigration.class).in(SINGLETON);
```
Finally, add it as a constructor parameter to the places where the database is used:
```java
@Inject
public MysqlDaoProvider(DbResourceGroupConfig config, @SuppressWarnings("unused") DatabaseReady ignored)
{ ... }
```
As part of this, I would actually change `MysqlDaoProvider` to take a `Jdbi` and create it once here, in the module, so that all the usages of `DatabaseReady` are together:
```java
@Provides
@Singleton
public static Jdbi create(DbResourceGroupConfig config, @SuppressWarnings("unused") DatabaseReady ignored)
{
    return Jdbi.create(config.getConfigDbUrl(), config.getConfigDbUser(), config.getConfigDbPassword());
}
```